src/cascadia/TerminalApp/BaseVisualStudioGenerator.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "BaseVisualStudioGenerator.h"
+#include "DefaultProfileUtils.h"
+
+std::vector<winrt::TerminalApp::Profile> TerminalApp::BaseVisualStudioGenerator::GenerateProfiles()
+{
+    std::vector<winrt::TerminalApp::Profile> profiles;
+
+    // There's no point in enumerating valid Visual Studio instances more than once,

```

Comment:
Instances can change most often without a reboot (most common cause of a reboot is the shared VC runtime which doesn't get updated often). Would it be better to put a timer on this to periodically check again? In the grand scheme of things, I suppose it's uncommon enough to not worry about it, but worth considering.

Reply:
This only caches once per run of terminal